arviz/plots/densityplot.py

```diff
@@ -135,9 +144,13 @@ def plot_density(
         >>> az.plot_density([centered, non_centered], var_names=["mu"], bw=.9)
     """
     if not isinstance(data, (list, tuple)):
-        datasets = [convert_to_dataset(data, group=group)]
-    else:
-        datasets = [convert_to_dataset(datum, group=group) for datum in data]
+        data = [data]
+
+    if coords is None:
+        coords = {}
+    datasets = [
+        get_coords(convert_to_dataset(datum, group=group), coords) for datum in data
```

Comment:
I think this will probably break if one of the inference data objects has a coordinate the other doesn't, which currently is one possibility in `plot_density`. Using inference data [`sel`](https://github.com/arviz-devs/arviz/blob/master/arviz/data/inference_data.py#L122) method instead of dataset `sel` will solve the issue with coordinate names by applying only the existing ones, but the case of mismatching coordinate values would not be solved. 

As I do not think this will be a common case, maybe the best option is to create a helper function in `densityplot.py` itself in order to take this 2 possibilities into account?

Reply:
I'm not sure what to do about this.  Maybe it would help to articulate what limits we put on the set of `InferenceData` objects we accept here?  Without doing something like that, I don't see how we can say whether or not this function will plot correctly.

To be honest, I'm not sure what is gained by permitting a set of `InferenceData` here instead of requiring the user to invoke this function multiple times, once per `InferenceData` object.  Presumably the issue is for model comparison, but I'm afraid I'm just guessing.

Because I don't know what is the answer to that, I don't know how to think about what we need to support here.